File: compiler/bench/ackmed.c
// expected: 8189
#include <stdio.h>
long ackmed(long m, long n) {
    if (m == 0) return n + 1;
    if (n == 0) return ackmed(m - 1, 1);
    return ackmed(m - 1, ackmed(m, n - 1));
}
int main() {
    printf("%ld\n", ackmed(3, 10));
    return 0;
}

File: compiler/bench/ackmed.fs
\ expected: 8189
\ Ackermann(3,10)

: ackmed ( m n -- r ) recursive
  over 0= if nip 1+ exit then
  dup 0= if drop 1- 1 recurse exit then
  over 1- -rot 1- recurse recurse ;

: main
  3 10 ackmed . cr ;
